libcob/fileio.c

```diff
@@ -757,6 +767,22 @@ bdb_savekey (cob_file *f, unsigned char *keyarea, unsigned char *record, int idx
 	return (int)f->keys[idx].field->size;
 }
 
+static void
+bdb_setkeycol (cob_file *f, int idx)
+{
+	struct indexed_file	*p = f->file;
+	const cob_field	*fld = f->keys[idx].field;
+	/* We only apply a collating sequence if the key is alphanumeric */
+	if (COB_FIELD_IS_ANY_ALNUM(fld) || COB_FIELD_IS_NUMDISP(fld)) {
+		DBT_SET_APP_DATA(&p->key, (void *)f->keys[idx].collating_sequence);
+	/* TODO: this should be done for national, when available *//*
+	} else if (COB_FIELD_IS_NATIONAL(fld)) {
```

Comment:
better use `#if 0 /* TODO ddd */` here, this leads to a better highlighting and less"that's a strange code here" analyzer warning